### PR TITLE
hq resubmit

### PR DESCRIPTION
**Pull request by User3574**
Fixes: #33 
Fixes: #36

### CODE REVIEW

File: src/common/arrayparser.rs [hunk 1]
```diff
@@ -1,25 +1,46 @@
...
             _ => Err("Invalid range"),
         },
     )(input)
 }
 
 fn p_array_def(input: &str) -> NomResult<ArrayDef> {
-    map(p_task_id_range, ArrayDef::new)(input)
+    map_res(p_task_id_ranges, |r| match r {
+        res if !is_overlapping(res.clone()) => Ok(ArrayDef::new(res)),
+        _ => Err("Ranges overlap"),
```
Comment (Kobzol) [inline src/common/arrayparser.rs:31]: @spirali Why should overlapping ranges cause an error? Maybe a warning could be printed, but it seems to me that an error is too strict.

Comment (spirali) [reply src/common/arrayparser.rs:31]: I prefer the error. I do not see a usecase.

Comment (Kobzol) [reply src/common/arrayparser.rs:31]: It seems to me that overlap is totally harmless here.

It's an artificial example, but if I want to e.g. resubmit every even task and also tasks from some range, I could just do
`0-1000:2, 500-600`. With this check I would need to split the intervals manually.

**Comment by spirali** [comment]
I have did some squashing and small refactoring. It is ready to be merged. Can you please (@Kobzol @User3574) rebase it on main? I am getting some conflicts wrt. to Nom error messages)

**Comment by Kobzol** [comment]
@User3574 Please check of the rebased version is correct.

**Pull request merged**